### PR TITLE
Modifications enabling ghost grid points

### PR DESCRIPTION
**Pull request by rcthomas**
The velocity grid allocation is modified so that ghost grid points are
allocated.  The full grid including ghost zones is 3 times the size of
the real grid, based on considerations from the geometry of the line
forming region.  The real grid is of size `real_vr_count` and the full
grid including ghost zones is of size `vr_count`.  So, when iterating
over wavelengths in the `tau` and `src` arrays, the right chunk size is
`vr_count`.

Additionally only real grid cells need to have associated angles for
computing mean intensity.  So we don't make allocations for angles in
the ghost points.

Further specifics:

* Comment struct, add real velocity grid variable
* Comments explain choice of 3 for full grid factor
* Comments explain why angles aren't allocated for ghosts
* Changes to the output format to include real grid size

### CODE REVIEW

**Review by bcfriesen — COMMENTED** [review]
The content is good. I would have preferred splitting up this one commit into several, as it makes a number of orthogonal changes. E.g., the following could be separate commits:

- the `uncrustify` changes made to the `printf()` lines 42-48 of `resist-context.c`
- all remaining changes in `resist-context.c`, as well as the changes to lines 22-23 of `resist-context.h`
- all remaining changes of `resist-context.h`

I think one element of the social contract we haven't agreed on up to this point is how mean to be about rejecting PRs that don't conform to the dev docs. So you are welcome either to merge this as is, or I can show you the fast way to "un-squash" it into the three separate commits mentioned above.

**Comment by rcthomas** [comment]
I'm not sure I agree on what you mean by orthogonal, especially where a header file and a source file are concerned.  What I would like is that at each commit the header and source files are coherent in terms of syntax (of course) and conceptually.  I changed a definition here, and for me to make a commit where I changed a definition but didn't propagate the implications of that change to the source file (from the header file) seems like I am splitting up covariant changes.

At this point I am thinking perhaps I should just never squash, then do the PR and have you tell me if I should squash, and then I will squash.  :)

**Comment by bcfriesen** [comment]
The reasons for segmenting the commits in the way I suggested are the following (listed in the same order as the commit suggestions themselves):

1. This change is the result of just running `make format` (or `make reformat`). It does not depend on the change in definition of `vr_count`.
2. All of these changes correspond to the change in definition of `vr_count`.
3. These comments are unrelated to the change in definition of `vr_count`.

**Comment by rcthomas** [comment]
Actually `uncrustify` isn't doing much for 1.  Note the format codes have been updated in `printf` and I added the new variable to the output. 

So let's just take that one as an example.  Is that 2 commits or one?  My thinking while I did it was "here's this new variable to include in the output, and while I'm at it let's set some field widths while were at it, that's not too big a deal is it?"  But... what's the answer in your mind?

**Comment by bcfriesen** [comment]
> Actually uncrustify isn't doing much for 1. Note the format codes have been updated in printf and I added the new variable to the output.

I see. To my mind that is two unrelated changes - one is adding output for the new variable, and the other is changing the output formatting for all variables. I would have split it up such that one commit added the output for the new variable with the old formatting, and a second commit which changed all formatting for all variables to the new format. Since you've already done it all in one go, it seems kind of stupid to go back and split it up in to one commit that does it "wrong" and then another commit that fixes it. For people who don't spend all day doing git gymnastics it becomes a huge waste of time, which is why I'm not pushing too hard to change what you did.

**Comment by rcthomas** [comment]
So, I am old, I mostly just edit code and then at some point go "ah that's enough changes" and then I look at the diffs and try to remember what I did, and then write a snappy commit message like "look what a genius I am" or something like that.  What it sounds like is that I should think about making a change, committing that one, and moving on to the next thing.  This obviously breaks down the flow, and it makes it easier for others to follow what I am thinking.  But honestly I don't really go in straight lines, and I think that ends up generating a bunch of nonsense in the history if I document it all.  Like I said I am used to either working alone all the time, or else my changes for the most part end up not mattering to other people when I collaborate on a bigger project (statistically speaking that's always true).

As this project gets bigger and there are more lines of code I can't get away with this obviously.  The unit of change is a commit, I am just used to having one level of change, not 2 (PR and commit) and I am not sure I can learn how to manage that but I will try.

Let me have this one (waiting for approval for this commit, do you have a button to hit on your side to approve) and I will keep working on getting our workflow right.  I will try to not squash everything all the time.

**Comment by bcfriesen** [comment]
Understandable. This one is yours. What is more important than having a rigid set of rules is having a social contract to which everyone agrees.

I can't figure out how to approve this. Originally I chose to "comment" on this PR rather than "approve" or "request changes". But maybe there is a way to change this "comment" to approve. One moment ...

**Review by bcfriesen — APPROVED** [review]
Tested on a Macbook Pro. Works as expected. The fight about commit granularity is in a parallel discussion, but we converged on agreement.

**Pull request merged**
